code/datums/components/radioactive.dm

```diff
@@ -0,0 +1,125 @@
+			COMSIG_ATOM_ENTERED,
+			COMSIG_ATTACKHAND,
+			COMSIG_ITEM_EQUIPPED,
+			COMSIG_ITEM_PICKUP,
+			COMSIG_MOB_GRABBED,
+			COMSIG_ITEM_ATTACK_POST,
+		), .proc/touched)
+		RegisterSignal(parent, list(COMSIG_ITEM_CONSUMED_PARTIAL, COMSIG_ITEM_CONSUMED_ALL), .proc/eaten)
+
+		if(isitem(parent))
+			RegisterSignal(parent, list(COMSIG_ITEM_PROCESS), .proc/ticked)
+			if(!(parent in global.processing_items))
+				global.processing_items.Add(parent)
+				src._added_to_items_processing = TRUE
+		else if(ismob(parent))
+			RegisterSignal(parent, list(COMSIG_LIVING_LIFE_TICK), .proc/ticked)
+		else
+			global.processing_items.Add(src) //gross - in the event that this component is put on something that isn't an item/mob, use the item processing loop anyway
+		var/atom/PA = parent
+		var/color = neutron ? "#2e3ae4FF" : "#18e022FF"
+		PA.add_filter("radiation_color_\ref[src]", 1, color_matrix_filter(normalize_color_to_matrix(PA.color ? PA.color : "#FFF")))
+		src.backup_color = PA.color
+		PA.color = null
+		PA.add_simple_light("radiation_light_\ref[src]", rgb2num(color)+list(min(128,round(255*radStrength/100))))
+		PA.add_filter("radiation_outline_\ref[src]", 2, outline_filter(size=1.3, color=color))
+
+	proc/process()
+		if(QDELETED(parent))
+			global.processing_items.Remove(src)
+			return
+		ticked(parent)
+
+	UnregisterFromParent()
+		. = ..()
+		var/atom/PA = parent
+		if(src._added_to_items_processing)
+			global.processing_items.Remove(parent)
+		global.processing_items.Remove(src)
+		PA.remove_simple_light("radiation_light_\ref[src]")
+		PA.remove_filter("radiation_outline_\ref[src]")
+		PA.remove_filter("radiation_color_\ref[src]")
+		PA.color = backup_color
+		UnregisterSignal(parent, list(COMSIG_ATOM_EXAMINE))
+		UnregisterSignal(parent, list(COMSIG_ATOM_CROSSED,
+			COMSIG_ATOM_ENTERED,
+			COMSIG_ATTACKHAND,
+			COMSIG_ITEM_EQUIPPED,
+			COMSIG_ITEM_PICKUP,
+			COMSIG_MOB_GRABBED,
+			COMSIG_ITEM_ATTACK_POST,
+		))
+		UnregisterSignal(parent, list(COMSIG_ITEM_CONSUMED_PARTIAL, COMSIG_ITEM_CONSUMED_ALL))
+		if(isitem(parent))
+			UnregisterSignal(parent, list(COMSIG_ITEM_PROCESS))
+		else if(ismob(parent))
+			UnregisterSignal(parent, list(COMSIG_LIVING_LIFE_TICK))
+
+	InheritComponent(datum/component/radioactive/R, i_am_original)
+		if (i_am_original)
+			src.radStrength = min(src.radStrength+R.radStrength, 100)
+			src.neutron |= R.neutron //only one type of radiation allowed, and neutron takes precedence
+			src.decays &= R.decays //permenant radiation takes precedence over decay
+
+	proc/ticked(atom/owner, mult=1)
+		for(var/mob/M in viewers(effect_range,parent))
+			M.take_radiation_dose(mult * (neutron ? 0.2 : 0.05) * (radStrength/100))
+		if(src.decays && prob(33))
+			src.radStrength = max(0, src.radStrength - mult)
+		if(!src.radStrength)
+			src.RemoveComponent()
+
+	proc/touched(atom/owner, mob/toucher)
+		if(istype(toucher))
+			toucher.take_radiation_dose((neutron ? 0.2 : 0.05) * (radStrength/100))
```

Comment:
Should probably put a short cooldown on this, unless potential doubledipping the proc on stuff like COMSIG_ATTACKHAND, COMSIG_ITEM_PICKUP is intentional.

Reply:
A little bit of double dipping is intentional, because standing next to multiple rad sources should be more deadly than standing next to one. I'll put a short (sub item tick time) cooldown on it anyway though, to stop something like dumping a stack of ore next to someone being instantly fatal.